### ISSUE TITLE
Allow customizing translate config used by DocumentClientV3  

### ISSUE THREAD

**Issue by bodhihawken** [typedorm/typedorm#319]
Not sure if this is pure stupidity on my behalf, but I've spent some time battling the AWS SDK V3 on undefined values for dynamoDB.
(A problem I didn't run into with V2)

And unfortunately, we don't yet expose the marshal options as part of SDK V3. 
(Manually changing removeUndefinedValues to true seems to fix all my issues)

Thoughts on allowing the options to be passed as part of constructing the document client?

Happy to put forward a PR for this.

**Comment by derelk** [comment]
I've been running into this too and was also thinking of exploring how to expose the marshal options.

**Comment by GTRanger** [comment]
It's been a while, but iirc that error is a red herring. Make sure your version of `class-transformer` matches exactly what is used by `typedorm`.

**Comment by GTRanger** [comment]
I believe this is a bug with TypeDORM. You have to ensure that you never pass keys with undefined values to the update body (i.e. if a value should be undefined the update body **should not** have that key). For example:

```
// BAD
{
  requiredProp: "hello",
  optionalProp: undefined
}

// GOOD
{
  requiredProp: "hello",
}

// ALSO GOOD
{
  requiredProp: "hello",
  optionalProp: "value"
}
```


You can use the following syntax to achieve this result:

```javascript
const updateBody = {...(props.optionalProp && { optionalProp: props.optionalProp })}
entityManager.update(item, primary, updateBody)
```

In the event that the optional prop is undefined then the expression will simply return `{}` rather than `{optionalProp: undefined}`. If it is not undefined you will get `{optionalProp: "value"}`

**Comment by smcroskey** [comment]
Would also love to see this supported.  We've worked around it by implementing our own document client, but it's a bit heavy-handed for needing to just override the one `removeUndefinedValues` property:


```ts
const DEFAULT_TRANSLATE_CONFIG: TranslateConfig = {
    marshallOptions: {
        // Whether to automatically convert empty strings, blobs, and sets to `null`.
        convertEmptyValues: false,
        // Whether to remove undefined values while marshalling.
        removeUndefinedValues: true,
        // Whether to convert typeof object to map attribute.
        convertClassInstanceToMap: false,
    },
    unmarshallOptions: {
        // Whether to return numbers as a string instead of converting them to native JavaScript numbers.
        wrapNumbers: false,
    },
};

/**
 * Simple wrapper over the DocumentClientV3 class from @typedorm that allows us to customize the TranslateConfig
 * used by the underlying DynamoDBDocumentClient.
 */
export class DocumentClient extends DocumentClientV3 {
    readonly documentClient: DynamoDBDocumentClient;

    constructor(dynamoDBClient: DynamoDBClient, translateConfig: TranslateConfig = DEFAULT_TRANSLATE_CONFIG) {
        super(dynamoDBClient);
        this.documentClient = DynamoDBDocumentClient.from(dynamoDBClient, translateConfig);
    }
}
```

**Comment by derekspelledcorrectly** [comment]
I think what @GTRanger is saying is that the marshaling error is a symptom of another problem: TypeDORM is marshaling objects that have keys with explicitly undefined values, rather than just omitting the key. The marshaler wouldn't complain if the keys were just omitted.

I didn't verify that behavior or know enough to know if it's true keys shouldn't have undefined as value, but it makes sense conceptually to omit keys with undefined values.

**Comment by hernanimattos** [comment]
I have the same problem, but we choice don't do the wrapper if change the version this could be a problem. But if these props are able in the constructor to resolve the problem.

**Comment by whimzyLive** [comment]
I'm exposing the translateConfig options to the DocumentClientV3 constructor, which will be made available in a future minor release - 1.16.

It should make working with DocumentClientV3 more convenient.  @smcroskey.

If the entity attributes `marshling/unmarshaling has uncovered any other issue, I recommend opening a new issue @derelk, @GTRanger  as this will be auto closed when the PR related to exposing `translateConfigOptoins` gets merged
